tests/test_node.py

```diff
@@ -99,13 +99,35 @@ def test_dfs_graph_with_pruning_transformations(self):
         n6 = node.Filter()
 
         n5 = n1.Count()
+        ## This node removes references from the filter
+        ## node defined against n5 above
 
         obtained_order = DfsTest.traverse(node = node._get_head())
 
         reqd_order = [1, 3, 2, 2, 3, 2]
 
         self.assertEqual(obtained_order, reqd_order)
 
+    def test_dfs_graph_with_recursive_pruning(self):
```

Comment:
Hi Shravan, this seems ok. Can you thing of any other scenarios we could test?
Do you think the following scenario is different enough from the ones we already have?
```
(...same code as in this function...)
n7 = n2.Count()
```
This tests that, even though `n2 = None`, it doesn't get cleaned because another node depends on it.
Diogo

Reply:
Yes @diocas . That'd a good test case. Thanks for the suggestion :)

Reply:
@diocas , added a new commit to test that :)